circuits/core/manager.py

```diff
@@ -219,7 +221,7 @@ def __repr__(self):
 
         channel = "/{0:s}".format(str(getattr(self, "channel", "")))
 
-        q = len(self._queue)
+        q = len(self)
```

Comment:
Yes, you two are absolutely right -- a custom object is probably the "most compatible"